From 0e80878132fa3176f25e564b1869faca038fd9af Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@polymtl.ca>
Date: Tue, 12 Aug 2008 10:32:22 -0400
Subject: [PATCH] lttng-instrumentation-kernel-tracepoints-probes-no-modules

LTTng instrumentation kernel tracepoint probes - no modules

Only define and call the module related functions if kernel is compiled with
module support. Will not compile otherwise.

Signed-off-by: Benjamin Poirier <benjamin.poirier@polymtl.ca>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@polymtl.ca>
---
 kernel/kernel-trace.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/kernel/kernel-trace.c b/kernel/kernel-trace.c
index 5d0bb93..eb6b8cd 100644
--- a/kernel/kernel-trace.c
+++ b/kernel/kernel-trace.c
@@ -209,6 +209,7 @@ static void probe_kernel_vprintk(unsigned long retaddr, char *buf, int len)
 	}
 }
 
+#ifdef CONFIG_MODULES
 static void probe_kernel_module_free(struct module *mod)
 {
 	trace_mark(kernel_module_free, "name %s", mod->name);
@@ -218,6 +219,7 @@ static void probe_kernel_module_load(struct module *mod)
 {
 	trace_mark(kernel_module_load, "name %s", mod->name);
 }
+#endif
 
 int __init kernel_trace_init(void)
 {
@@ -288,10 +290,12 @@ int __init kernel_trace_init(void)
 	WARN_ON(ret);
 	ret = register_trace_kernel_vprintk(probe_kernel_vprintk);
 	WARN_ON(ret);
+#ifdef CONFIG_MODULES
 	ret = register_trace_kernel_module_free(probe_kernel_module_free);
 	WARN_ON(ret);
 	ret = register_trace_kernel_module_load(probe_kernel_module_load);
 	WARN_ON(ret);
+#endif
 
 	return 0;
 }
@@ -300,8 +304,10 @@ module_init(kernel_trace_init);
 
 void __exit kernel_trace_exit(void)
 {
+#ifdef CONFIG_MODULES
 	unregister_trace_kernel_module_load(probe_kernel_module_load);
 	unregister_trace_kernel_module_free(probe_kernel_module_free);
+#endif
 	unregister_trace_kernel_vprintk(probe_kernel_vprintk);
 	unregister_trace_kernel_printk(probe_kernel_printk);
 
-- 
1.5.5.1

